llvm/Makefile.am

```diff
@@ -1,30 +1,23 @@
-#
-# Conditional submodule for llvm
-#
-# make reset-llvm will checkout a version of llvm which is suitable for this version of mono
-# into $top_srcdir/llvm/llvm.
-#
 
-LLVM_PATH=llvm
 
-SUBMODULES_CONFIG_FILE = $(top_srcdir)/llvm/SUBMODULES.json
-include $(top_srcdir)/scripts/submodules/versions.mk
+EXTRA_DIST=SUBMODULES.json
 
-$(eval $(call ValidateVersionTemplate,llvm,LLVM))
+if INTERNAL_LLVM
+.PHONY: all-local
+all-local: configure-llvm build-llvm install-llvm
 
-# Bump the given submodule to the revision given by the REV make variable
-# If COMMIT is 1, commit the change
-bump-llvm: __bump-version-llvm
+.PHONY: clean-local
+clean-local: clean-llvm
+endif
 
-# Bump the given submodule to the branch given by the BRANCH/REMOTE_BRANCH make variables
-# If COMMIT is 1, commit the change
-bump-branch-llvm: __bump-branch-llvm
+if INTERNAL_LLVM_ASSERTS
+llvm_build_args := "INTERNAL_LLVM_ASSERTS=yes"
+endif
 
-# Bump the given submodule to its current GIT version
-# If COMMIT is 1, commit the change
-bump-current-llvm: __bump-current-version-llvm
+# We do not propagate top level configuration variables
+# to every makefile in the repo. Since build.mk is not going to become
+# an automake file, I'm passing the variables explicitly.
 
-clean-local:
-	$(RM) -r $(LLVM_PATH)
+reset-llvm configure-llvm build-llvm install-llvm clean-llvm: build.mk
```

Comment:
Why not just including `build.mk`?

Reply:
Because this is an automake file, not a makefile. If you include build.mk, you get all kinds of failures when trying to build because build.mk has a lot of syntax that is not valid automake.